src/Symfony/Component/Validator/Constraints/WhenValidator.php

```diff
@@ -33,6 +33,8 @@ public function validate(mixed $value, Constraint $constraint): void
         $variables = $constraint->values;
         $variables['value'] = $value;
         $variables['this'] = $context->getObject();
+        $variables['context'] = $context;
+        $variables['root'] = $context->getRoot();
```

Comment:
Not sure we need `root`, as we can access it with `context.getRoot()` in the expression if we inject `context`.